lib/moonshot/tools/asg_rollout.rb

```diff
@@ -65,6 +65,10 @@ def wait_for_in_service(new_instance)
 
           loop do
             instance_health = asg.instance_health(new_instance)
+            if instance_health.out_of_service?
```

Comment:
The instance will always start in OutOfService, before it (hopefully) moves into InService. What we might want to do here is implement a timeout, and if we've passed that timeout and we're in either OutOfService state *or* missing (which `out_of_service?` may cover?), then abort.